src/main/java/org/folio/rest/impl/PatronServicesResourceImpl.java

```diff
@@ -288,35 +291,23 @@ public void postPatronAccountHoldCancelByIdAndHoldId(String id, String holdId, H
 
     final Hold[] holds = new Hold[1];
 
-    try {
-      httpClient.get("/circulation/requests/" + holdId, Map.of(), okapiHeaders)
-        .thenApply(ResponseInterpreter::verifyAndExtractBody)
-        .thenApply( body -> {
-          final Item item = getItem(body);
-          final Hold hold = getHold(body, item);
-          holds[0] = constructNewHoldWithCancellationFields(hold, entity);
-          return body;
-        })
-        .thenCompose( anUpdatedRequest -> {
-          try {
-            JsonObject cancelRequest = createCancelRequest(anUpdatedRequest, entity);
-            return httpClient.put("/circulation/requests/" + holdId, cancelRequest, okapiHeaders);
-          } catch (Exception e) {
-              asyncResultHandler.handle(handleHoldCancelPOSTError(e));
-              return null;
-            }
-        })
-        .thenApply(ResponseInterpreter::verifyAndExtractBody)
-        .thenAccept(
-            body -> asyncResultHandler.handle(succeededFuture(
-              PostPatronAccountHoldCancelByIdAndHoldIdResponse.respond200WithApplicationJson(holds[0]))))
-        .exceptionally(throwable -> {
-            asyncResultHandler.handle(handleHoldCancelPOSTError(throwable));
-            return null;
-        });
-    } catch (Exception e) {
-      asyncResultHandler.handle(handleHoldCancelPOSTError(e));
-    }
+    completedFuture("/circulation/requests/" + holdId)
+      .thenCompose(path -> httpClient.get(path, Map.of(), okapiHeaders))
+      .thenApply(ResponseInterpreter::verifyAndExtractBody)
+      .thenApply(body -> {
+        holds[0] = constructNewHoldWithCancellationFields(getHold(body, getItem(body)), entity);
```

Comment:
Why do we need this `holds` array? Its size is always 1, why not just store this one element in a variable?